docs/reference/search/search-your-data/retrieve-selected-fields.asciidoc

```diff
@@ -412,9 +368,113 @@ flag hasn't been set to `true` for that field pattern.
 // TESTRESPONSE[s/"max_score" : 1.0/"max_score" : $body.hits.max_score/]
 // TESTRESPONSE[s/"_score" : 1.0/"_score" : $body.hits.hits.0._score/]
 
+[discrete]
+[[source-filtering]]
+=== The `_source` option
+
+You can use the `_source` parameter to select what fields of the source are
+returned. This is called _source filtering_.
+
+The following search API request sets the `_source` request body parameter to
+`false`. The document source is not included in the response.
+
+[source,console]
+----
+GET /_search
+{
+  "_source": false,
+  "query": {
+    "match": {
+      "user.id": "kimchy"
+    }
+  }
+}
+----
+
+To return only a subset of source fields, specify a wildcard (`*`) pattern in
+the `_source` parameter. The following search API request returns the source for
+only the `obj` field and its properties.
+
+[source,console]
+----
+GET /_search
+{
+  "_source": "obj.*",
+  "query": {
+    "match": {
+      "user.id": "kimchy"
+    }
+  }
+}
+----
+
+You can also specify an array of wildcard patterns in the `_source` field. The
+following search API request returns the source for only the `obj1` and
+`obj2` fields and their properties.
+
+[source,console]
+----
+GET /_search
+{
+  "_source": [ "obj1.*", "obj2.*" ],
+  "query": {
+    "match": {
+      "user.id": "kimchy"
+    }
+  }
+}
+----
+
+For finer control, you can specify an object containing arrays of `includes` and
+`excludes` patterns in the `_source` parameter.
+
+If the `includes` property is specified, only source fields that match one of
+its patterns are returned. You can exclude fields from this subset using the
+`excludes` property.
+
+If the `includes` property is not specified, the entire document source is
+returned, excluding any fields that match a pattern in the `excludes` property.
+
+The following search API request returns the source for only the `obj1` and
+`obj2` fields and their properties, excluding any child `description` fields.
+
+[source,console]
+----
+GET /_search
+{
+  "_source": {
+    "includes": [ "obj1.*", "obj2.*" ],
+    "excludes": [ "*.description" ]
+  },
+  "query": {
+    "term": {
+      "user.id": "kimchy"
+    }
+  }
+}
+----
+
+[discrete]
+[[field-retrieval-methods]]
+=== Other methods of retrieving data
+A document's `_source` is stored as a single field in Lucene. So the whole
+`_source` object must be loaded and parsed even if only a small number of
+fields are requested. To avoid this limitation, you can try other options for
+loading fields:
+
+* Use the <<docvalue-fields, `docvalue_fields`>>
+parameter to get values for selected fields. This can be a good
+choice when returning a fairly small number of fields that support doc values,
+such as keywords and dates.
+* Use the <<request-body-search-stored-fields, `stored_fields`>> parameter to
+get the values for specific stored fields (fields that use the
+<<mapping-store,`store`>> mapping option).
```

Comment:
Maybe add a little note about how this is usually not required. Hell, maybe this whole section should have a note that usually you don't need any of these. maybe its enough to day "usually you don't have to, but if you absolutely need to force loading a stored or doc values field, use this. But usually `fields` is better."

Reply:
+1 to emphasizing that these are not usually required. I also have a personal TODO to update the docs around stored fields to de-emphasize them even more, as there are almost always more appropriate options.

Reply:
What if we add a subsection after this list that says:

> .Using `fields` is typically better
> ****
> These options are usually not required. Using the `fields` option is typically
> the better choice, unless you absolutely need to force loading a stored or
> `docvalue_fields`.
> ****

Reply:
I might put this at the top of the section to be honest. If I'm reading top to bottom its nice to have the perspective that I can generally ignore this section unless I have a "unique" use case.

Reply:
++

Good point -- I'll move that text to the beginning of the section.